Bottom: 99d930254eec97912edf5f76c08e8786bb36f25d
Top:    c65826557a47314de9efdbf436128ec22315a308
Author: Jon Smirl <jonsmirl@gmail.com>
Date:   2009-04-28 22:11:12 -0400

Refresh of temp1

---

diff --git a/sound/soc/fsl/mpc5200_dma.c b/sound/soc/fsl/mpc5200_dma.c
index ce45799..8ffa2fb 100644
--- a/sound/soc/fsl/mpc5200_dma.c
+++ b/sound/soc/fsl/mpc5200_dma.c
@@ -369,13 +369,23 @@ psc_dma_pointer(struct snd_pcm_substream *substream)
 	return frames + delta;
 }
 
+static int
+psc_dma_hw_params(struct snd_pcm_substream *substream,
+			 struct snd_pcm_hw_params *params)
+{
+  	snd_pcm_set_runtime_buffer(substream, &substream->dma_buffer);
+
+  	return 0;
+}
+
 static struct snd_pcm_ops psc_dma_ops = {
 	.open		= psc_dma_open,
 	.close		= psc_dma_close,
 	.hw_free	= psc_dma_hw_free,
 	.ioctl		= snd_pcm_lib_ioctl,
 	.pointer	= psc_dma_pointer,
-	.trigger	= psc_dma_trigger
+	.trigger	= psc_dma_trigger,
+	.hw_params	= psc_dma_hw_params,
 };
 
 static u64 psc_dma_dmamask = 0xffffffff;
diff --git a/sound/soc/fsl/mpc5200_psc_ac97.c b/sound/soc/fsl/mpc5200_psc_ac97.c
index 5663187..cc383a5 100644
--- a/sound/soc/fsl/mpc5200_psc_ac97.c
+++ b/sound/soc/fsl/mpc5200_psc_ac97.c
@@ -181,8 +181,6 @@ static int psc_ac97_hw_analog_params(struct snd_pcm_substream *substream,
 	else
 		out_be32(&psc_dma->psc_regs->ac97_slots, 0x03000000);
 
-  	snd_pcm_set_runtime_buffer(substream, &substream->dma_buffer);
-
 	return 0;
 }
 
diff --git a/sound/soc/fsl/mpc5200_psc_i2s.c b/sound/soc/fsl/mpc5200_psc_i2s.c
index c050a1f..dd74a24 100644
--- a/sound/soc/fsl/mpc5200_psc_i2s.c
+++ b/sound/soc/fsl/mpc5200_psc_i2s.c
@@ -107,9 +107,6 @@ static int psc_i2s_hw_params(struct snd_pcm_substream *substream,
 		out_be32(&psc_dma->psc_regs->ccr, value);
 		out_8(&psc_dma->psc_regs->ctur, bits - 1);
 	}
-
-  	snd_pcm_set_runtime_buffer(substream, &substream->dma_buffer);
-
 	return 0;
 }
